src/components/SinglePayment.js

```diff
@@ -1,16 +1,23 @@
 import React from "react";
 import Button from "./Button";
 
-const SinglePayment = ({ payment }) => {
+const SinglePayment = ({ payment, index, setPaymentData, paymentData, allPaymentData }) => {
```

Comment:
NOTE: In real case scenario, index will be part of `paymentData` so you won't have to pass it down like this ;)